### PR TITLE
vt-doc: Add a Troubleshooting section

### PR DESCRIPTION
**Pull request by jfehlig**
### PR creator: Description

The beginnings of a troubleshooting section for the Virtualization
Guide. More info can be added later, for example items sepcific to
Xen and KVM.Describe the overall goals of this pull request.

!!!!!NOTE: when backporting to 15 SP3, update the external link to supportconfig in the Admin guide!!!!!

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 SP4/openSUSE Leap 15.4 *(current `main`, no backport necessary)*
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by jfehlig** [comment]
FYI, @hw-claudio

File: xml/virt_docs.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE chapter
+[
+  <!ENTITY % entities SYSTEM "generic-entities.ent">
+    %entities;
+]>
+
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0" xml:id="cha-virt-docs">
+ <title>Help and documentation</title>
```
Comment (hw-claudio) [inline xml/virt_docs.xml:9]: Maybe "Integrated Help and package documentation?" or something like that?
Plain "documentation" feels a bit a recursive self-reference to this very document.

File: xml/virt_docs.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+   <dm:translation>yes</dm:translation>
+  </dm:docmanager>
+ </info>
+ <para>
+  Both &xen; and &kvm; hosts include various sources of information and documentation,
+  many of which are already available your installed system. The various documentation
```
Comment (hw-claudio) [inline xml/virt_docs.xml:18]: "available in"..

File: xml/virt_docs.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0" xml:id="cha-virt-docs">
+ <title>Help and documentation</title>
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+   <dm:translation>yes</dm:translation>
+  </dm:docmanager>
+ </info>
+ <para>
+  Both &xen; and &kvm; hosts include various sources of information and documentation,
```
Comment (hw-claudio) [inline xml/virt_docs.xml:17]: Both xen and kvm host installations include out-of-the-box (is this the meaning you want to convey?) help integrated in the command line tools and in the virtualization packages.

File: xml/virt_docs.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ </para>
+ <variablelist>
+  <varlistentry>
+   <term>Integrated help and man pages for shell commands</term>
+   <listitem>
+    <para>
+     The various virtualization packages provide commands for managing many aspects
+     of the virtualization host. It is not possible or expected to remember all
+     options supported by these commands. Most commands include integrated 'help'
+     subcommands and man pages which describe the command and any supported options.
```
Comment (hw-claudio) [inline xml/virt_docs.xml:29]: should we add an example? I would suggest virsh or/and virt-install:

"For example, to install a guest from an OS installation image, the command virt-install is available.

Type:

$ man virt-install

to get detailed information about the program use and all available options. You can find the section labeled "EXAMPLES" towards the end of the man page, with some useful command line examples.

For a more succinct description of available commands and options, run the program as follows:

$ virt-install --help

For tools that can be used interactively, like the "virsh" command,
in addition to 

$ man virsh

and

$ virsh --help

it is also possible to get help out a specific command in the middle of an interactive session, as such:

$ virsh
Welcome to virsh, the virtualization interactive terminal.

Type:  'help' for help with commands
       'quit' to quit

virsh # help
...(to get the list of commands)...

virsh # help start
... (to show the help specific to the "start" command)

File: xml/virt_docs.xml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+   </listitem>
+  </varlistentry>
+  <varlistentry>
+   <term>Documentation in <filename>/usr/share/doc/packages</filename></term>
+   <listitem>
+     <para>
+      Many of the virtualization packages provide additional content in their
+      documentation sub-package. As an example, the libvirt-doc package contains
+      all of the documentation available at libvirt.org, plus sample code
+      demonstrating use of the libvirt C API.
```
Comment (hw-claudio) [inline xml/virt_docs.xml:40]: maybe we can add:

"You can find the libvirt-doc package documentation in /usr/share/doc/packages/" (XXX note I see only README.rst NEWS.rst AUTHORS.rst there XXX),

and you can find the full documentation from libvirt.org in /usr/share/doc/libvirt/html/ .

Examples, including XML VM configuration snippets, can be found in /usr/share/doc/libvirt/examples/.

File: xml/virt_logs.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  <command>supportconfig</command> tool. See <xref linkend="sec-admsupport-supportconfig-sr"/>
+  for more information about <command>supportconfig</command>.
+ </para>
+ <para>
+  In some cases the information gathered by <command>supportconfig</command> is
+  insufficient and logs generated from a custom logging or debugging configuration
+  might be required to determine the cause of a problem.
+ </para>
+ <sect1 xml:id="sec-logs-libvirt">
+  <title>&libvirt; log controls</title>
```
Comment (hw-claudio) [inline xml/virt_logs.xml:28]: I think we want to extend this to answer the user question:

1) Where do I find the libvirt logs in an "out of the box" configuration?
 
2) How do I change the logging configuration?

3) In which cases is more logging required, and how do I properly configure it?

(I would go simplest use case first: logging everything INFO+ to a file seems immediately actionable and useful)

4) how do I configure more advanced logging? (Filters etc) -> separate Advanced Logging section

**Review by hw-claudio — COMMENTED** [review]
I'll look further into the advanced logging topics (filters etc)

File: xml/virt_logs.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE chapter
+[
+  <!ENTITY % entities SYSTEM "generic-entities.ent">
+    %entities;
+]>
+
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0" xml:id="cha-virt-logs">
+ <title>Gathing system information and logs</title>
```
Comment (hw-claudio) [inline xml/virt_logs.xml:9]: I would separate these two.
Users might be interested in collecting logs independently from a troubleshooting session with SUSE.

So I'd have <title>Gathering System Information for Virtualization Host troubleshooting</tile>
where we  would talk about supportconfig, mentioning that often collecting from both host and guest can be useful when possible etc, and then for the logging mentioning to also include libvirt log, and the libvirt driver log, and make a reference to the following section.

Comment (hw-claudio) [reply xml/virt_logs.xml:9]: maybe it already is, and I am confused about the tags.

File: xml/book_virtualization.xml [hunk 8]
```diff
@@ -112,6 +112,16 @@
   <xi:include href="qemu_running_vms_qemukvm.xml"/>
   <xi:include href="qemu_monitor.xml"/>
  </part>
+
+<!-- ===================================================================== -->
+<!-- Troubleshooting part                                                             -->
+<!-- ===================================================================== -->
+ <part xml:id="part-virt-troubleshoot">
+  <title>Troubleshooting</title>
+  <xi:include href="virt_docs.xml"/>
```
[the remark this answers is not in the document]
Comment (jfehlig) [reply xml/book_virtualization.xml:121]: Yes, nice suggestion!

File: xml/book_virtualization.xml [hunk 9]
```diff
@@ -112,6 +112,16 @@
...
+  <xi:include href="virt_logs.xml"/>
+ </part>
+
 <!-- ===================================================================== -->
 <!-- LXC part                                                              -->
```
[the remark this answers is not in the document]
Comment (jfehlig) [reply xml/book_virtualization.xml:126]: That's just part of the diff context. Nothing related to LXC has been touched by this change. And if more context was visible you would see the LXC doc is commented out.

File: xml/virt_docs.xml (hunk 3 above)
Comment (jfehlig) [reply xml/virt_docs.xml:17]: I've improved the intro paragraph.

File: xml/virt_docs.xml (hunk 4 above)
Comment (jfehlig) [reply xml/virt_docs.xml:29]: At the risk of needing to add more examples for the *many* commands we have, I added some for virt-install. I'd think any reasonable sysadmin knows how to use man pages and --help option.

File: xml/virt_docs.xml (hunk 5 above)
Comment (jfehlig) [reply xml/virt_docs.xml:40]: I reworked this paragraph a bit and think it addresses your comment in a more general way. But I'll let you be the judge of that once I push an updated version :-).

File: xml/virt_logs.xml (hunk 7 above)
Comment (jfehlig) [reply xml/virt_logs.xml:9]: The supportconfig tool already grabs everything we need. If it is missing something, we fix it. IMO it is sufficient to mention the tool as a way to collect all needed info. For tougher bugs, we may need to tweak the system and collect additional info. The "logs" part is about that.

File: xml/virt_logs.xml (hunk 6 above)
Comment (jfehlig) [reply xml/virt_logs.xml:28]: I improved the logging section and hopefully answered these types of user questions.

File: xml/virt_docs.xml (hunk 1 above)
Comment (jfehlig) [reply xml/virt_docs.xml]: Reworded this a bit. Thanks for the suggestion!

**Comment by jfehlig** [comment]
@tbazant, I'm done with the initial content. More can be added later.

One of the pipeline checks fails since I don't have the reference to supportconfig in the admin book quite right. Maybe you can help with that one, and any other formatting issues.

File: xml/virt_logs.xml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
```
Comment (tbazant) [inline xml/virt_logs.xml:9]:
```suggestion
 <title>Gathering system information and logs</title>
```

File: xml/virt_logs.xml [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+ <sect1 xml:id="sec-logs-libvirt">
+  <title>&libvirt; log controls</title>
+
+  <para>
+   &libvirt; provides logging facilities for both the library and the daemon.
+   Behavior of the logging facility is controlled by adjusting the log level,
+   filter, and output settings.
+  </para>
+  
+  <itemizedlist mark="bullet" spacing="normal">
```
Comment (tbazant) [inline xml/virt_logs.xml:36]:
```suggestion
  <itemizedlist>
```

File: xml/virt_logs.xml [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+   <dm:translation>yes</dm:translation>
+  </dm:docmanager>
+ </info>
+ <para>
+  When a virtualization host encounters a problem it is often necessary to
+  collect a detailed system report, which can be done with the help of the
+  <command>supportconfig</command> tool. See <xref linkend="cha-adm-support"/>
```
Comment (tbazant) [inline xml/virt_logs.xml:19]:
```suggestion
  <command>supportconfig</command> tool. See <link xlink:href="https://documentation.suse.com/sles/15-SP4/html/SLES-all/cha-adm-support.html"/>
```

File: xml/virt_logs.xml [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+    <formalpara>
+     <title>Log outputs</title>
+     <para>
+      Log outputs allow specifying where the filtered log messages are sent.
+      Messages can be sent to a file, the standard error stream of the process,
+      or journald. By default filtered log messages are sent to journald.
+     </para>
+    </formalpara>
+   </listitem>
+  </itemizedlist>
```
Comment (tbazant) [inline xml/virt_logs.xml:70]:
```suggestion
<variablelist>
 <varlistentry>
  <term>Log level</term>
  <listitem>
   <para>
    &libvirt; log messages are classified into four priority levels: DEBUG,
    INFO, WARNING, and ERROR. The DEBUG level is very verbose and capable of
    generating gigabytes of information in a short time. The volume of log
    messages becomes progressively less with the INFO, WARNING, and ERROR log
    levels. ERROR is the default log level.
   </para>
  </listitem>
 </varlistentry>
 <varlistentry>
  <term>Log filters</term>
  <listitem>
   <para>
    Log filters provide a way to log only messages matching a specific
    component and log level. Log filters allow collecting the verbose DEBUG
    log messages of specific components, but only ERROR level log messages from
    the rest of the system. By default, no log filters are defined.
   </para>
  </listitem>
 </varlistentry>
 <varlistentry>
  <term>Log outputs</term>
  <listitem>
   <para>
    Log outputs allow specifying where the filtered log messages are sent.
    Messages can be sent to a file, the standard error stream of the process,
    or journald. By default, filtered log messages are sent to journald.
   </para>
  </listitem>
 </varlistentry>
</variablelist>
```

**Review by tbazant — CHANGES_REQUESTED** [review]
Consider my suggestions please :-)

File: xml/virt_help.xml [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+  <varlistentry>
+   <term>Man pages for shell commands</term>
+   <listitem>
+    <para>
+     Most commands include a man page that provides detailed information about
+     the command, describes any options, and in some cases gives example command
+     usage. For example, to see the manual for the <command>virt-install</command>
+     command type
+    </para>
+<screen>man virt-install</screen>
```
Comment (tbazant) [inline xml/virt_help.xml:34]:
```suggestion
<screen>&prompt.user;man virt-install</screen>
```

File: xml/virt_help.xml [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+  </varlistentry>
+  <varlistentry>
+   <term>Integrated help for shell commands</term>
+   <listitem>
+    <para>
+     Commands also include integrated help, providing more compact and topic-driven
+     documentation. For example to see a brief description of the
+     <command>virt-install</command> command type
+    </para>
+    <screen>virt-install --help</screen>
```
Comment (tbazant) [inline xml/virt_help.xml:45]:
```suggestion
    <screen>&prompt.user;virt-install --help</screen>
```

File: xml/virt_help.xml [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+    <para>
+     Integrated help can also be used to see the details of a specific option. For
+     example, to see the sub-options supported by the disk option type
+    </para>
+<screen>virt-install --disk help</screen>
```
Comment (tbazant) [inline xml/virt_help.xml:50]:
```suggestion
<screen>&prompt.user;virt-install --disk help</screen>
```

File: xml/virt_logs.xml [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+   See <link xlink:href="https://libvirt.org/logging.html"/> for more details
+   on &libvirt;'s log controls.
+  </para>
+
+  <para>
+   A default &libvirt; installation has log level set to ERROR, no log filters
+   defined, and log outputs set to journald. Log messages from the &libvirt;
+   daemon can be viewed with the <command>journalctl</command> command:
+  </para>
+<screen>jounalctl --unit libvirtd</screen>
```
Comment (tbazant) [inline xml/virt_logs.xml:81]:
```suggestion
<screen>&prompt.root;jounalctl --unit libvirtd</screen>
```

File: xml/virt_logs.xml [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+<screen>
+   log_filters="1:qemu.qemu_monitor_json"
+   log_outputs="1:file:/tmp/libvirtd.log"
+</screen>
+  <para>
+   Log controls for the &libvirt; daemon can be found in
+   <filename>/etc/libvirt/libvirtd.conf</filename>. The daemon must be
+   restarted after making any changes to the configuration file.
+  </para>
+<screen>systemctl restart libvirtd.service</screen>
```
Comment (tbazant) [inline xml/virt_logs.xml:101]:
```suggestion
<screen>&prompt.root;systemctl restart libvirtd.service</screen>
```

**Review by tbazant — CHANGES_REQUESTED** [review]
Added prompts to commands

**Review by tbazant — APPROVED** [review]
LGTM

File: xml/virt_help.xml [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+<screen>&prompt.user;virt-install --disk help</screen>
+   </listitem>
+  </varlistentry>
+  <varlistentry>
+   <term>Documentation sub-packages</term>
+   <listitem>
+    <para>
+     Many of the virtualization packages provide additional content in their
+     documentation sub-package. As an example, the <package>libvirt-doc</package> package contains
+     all of the documentation available at <link xlink:href="http://libvirt.org"/>, plus sample code
```
Comment (dariavladykina) [inline xml/virt_help.xml:59]:
```suggestion
     all the documentation available at <link xlink:href="http://libvirt.org"/>, plus sample code
```

File: xml/virt_help.xml [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+     demonstrating use of the libvirt C API. Use the <command>rpm</command> command
```
Comment (dariavladykina) [inline xml/virt_help.xml:60]:
```suggestion
     demonstrating the use of the libvirt C API. Use the <command>rpm</command> command
```

File: xml/virt_help.xml [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+     to view the contents of a documentation sub-package. For example to see the
```
Comment (dariavladykina) [inline xml/virt_help.xml:61]:
```suggestion
     to view the contents of a documentation sub-package. For example, to see the
```

File: xml/virt_help.xml [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+     contents of <package>libvirt-doc</package>
```
Comment (dariavladykina) [inline xml/virt_help.xml:62]:
```suggestion
     contents of <package>libvirt-doc</package>:
```

File: xml/virt_logs.xml [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0" xml:id="cha-virt-logs">
+ <title>Gathering system information and logs</title>
+ <info>
+  <dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+   <dm:bugtracker></dm:bugtracker>
+   <dm:translation>yes</dm:translation>
+  </dm:docmanager>
+ </info>
+ <para>
+  When a virtualization host encounters a problem it is often necessary to
```
Comment (dariavladykina) [inline xml/virt_logs.xml:17]:
```suggestion
  When a virtualization host encounters a problem, it is often necessary to
```

File: xml/virt_logs.xml [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+  collect a detailed system report, which can be done with the help of the
+  <command>supportconfig</command> tool. See <link xlink:href="https://documentation.suse.com/sles/15-SP4/html/SLES-all/cha-adm-support.html"/>
+  for more information about <command>supportconfig</command>.
+ </para>
+ <para>
+  In some cases the information gathered by <command>supportconfig</command>
```
Comment (dariavladykina) [inline xml/virt_logs.xml:23]:
```suggestion
  In some cases, the information gathered by <command>supportconfig</command>
```

File: xml/virt_logs.xml [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+  is insufficient and logs generated from a custom logging or debugging
```
Comment (dariavladykina) [inline xml/virt_logs.xml:24]:
```suggestion
  is insufficient, and logs generated from a custom logging or debugging
```

File: xml/virt_logs.xml [hunk 26]
```diff
@@ -0,0 +1,103 @@
...
+  configuration might be required to determine the cause of a problem.
+ </para>
+ <sect1 xml:id="sec-logs-libvirt">
+  <title>&libvirt; log controls</title>
+
+  <para>
+   &libvirt; provides logging facilities for both the library and the daemon.
+   Behavior of the logging facility is controlled by adjusting the log level,
```
Comment (dariavladykina) [inline xml/virt_logs.xml:32]:
```suggestion
   The behavior of the logging facility is controlled by adjusting the log level,
```

File: xml/virt_logs.xml [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+  
+<variablelist>
+ <varlistentry>
+  <term>Log level</term>
+  <listitem>
+   <para>
+    &libvirt; log messages are classified into four priority levels: DEBUG,
+    INFO, WARNING, and ERROR. The DEBUG level is very verbose and capable of
+    generating gigabytes of information in a short time. The volume of log
+    messages becomes progressively less with the INFO, WARNING, and ERROR log
```
Comment (dariavladykina) [inline xml/virt_logs.xml:44]:
```suggestion
    messages progressively decreases with the INFO, WARNING, and ERROR log
```

File: xml/virt_logs.xml [hunk 28]
```diff
@@ -0,0 +1,103 @@
...
+   defined, and log outputs set to journald. Log messages from the &libvirt;
+   daemon can be viewed with the <command>journalctl</command> command:
+  </para>
+<screen>&prompt.root;jounalctl --unit libvirtd</screen>
+  <para>
+   The default log facility settings are fine for normal operations and
+   provide useful messages for applications and users of &libvirt;, but
+   internal issues often require debug level messages. As an example,
+   consider a potential bug in the interaction between &libvirt; and the
+   &qemu; monitor. In this case we only need to see the debug messages of
```
Comment (dariavladykina) [inline xml/virt_logs.xml:87]:
```suggestion
   &qemu; monitor. In this case, we only need to see the debug messages of
```

File: xml/virt_logs.xml [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+   the communication between &libvirt; and &qemu;. The follow example
```
Comment (dariavladykina) [inline xml/virt_logs.xml:88]:
```suggestion
   the communication between &libvirt; and &qemu;. The following example
```

File: xml/virt_logs.xml [hunk 30]
```diff
@@ -0,0 +1,103 @@
...
+  </listitem>
+ </varlistentry>
+</variablelist>
+  <para>
+   See <link xlink:href="https://libvirt.org/logging.html"/> for more details
+   on &libvirt;'s log controls.
+  </para>
+
+  <para>
+   A default &libvirt; installation has log level set to ERROR, no log filters
```
Comment (dariavladykina) [inline xml/virt_logs.xml:77]:
```suggestion
   A default &libvirt; installation has the log level set to ERROR, no log filters
```

File: xml/virt_logs.xml [hunk 31]
```diff
@@ -0,0 +1,103 @@
...
+   defined, and log outputs set to journald. Log messages from the &libvirt;
+   daemon can be viewed with the <command>journalctl</command> command:
+  </para>
+<screen>&prompt.root;jounalctl --unit libvirtd</screen>
+  <para>
+   The default log facility settings are fine for normal operations and
+   provide useful messages for applications and users of &libvirt;, but
+   internal issues often require debug level messages. As an example,
```
Comment (dariavladykina) [inline xml/virt_logs.xml:85]:
```suggestion
   internal issues often require DEBUG level messages. As an example,
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please take a look at some corrections.

File: xml/virt_logs.xml [hunk 32]
```diff
@@ -0,0 +1,103 @@
...
+ <sect1 xml:id="sec-logs-libvirt">
+  <title>&libvirt; log controls</title>
+
+  <para>
+   &libvirt; provides logging facilities for both the library and the daemon.
+   Behavior of the logging facility is controlled by adjusting the log level,
+   filter, and output settings.
+  </para>
+  
+  <itemizedlist>
```
[the remark this answers is not in the document]
Comment (tbazant) [reply xml/virt_logs.xml]: DONE

**Pull request merged**